### ISSUE TITLE
[Android] Cheats: pasting text doesn't work, among others 

### ISSUE THREAD

**Issue by Gonckzilla** [flyinghead/flycast#871]
The experience of trying to add a cheat in text directly through the app is currently frustrating: you can't paste text, which means you have to manually type your cheat, many times (and here's the other issue) not being able to see what you type because 1. the keyboard keeps being summoned (meaning you can't hide it) 2. the code input box is not scrollable 3. it doesn't dynamically adjust size to account for the keyboard's height.
Also, the Android text pointer doesn't appear.
Then when saving, an error will come up saying it couldn't be saved, but the cheat actually will save correctly - for the session at least; it will not be there when starting the game again.

I believe using the native android keyboard box would fix most of these issues. I'm not sure what the term for this "module" is but I'll provide a video. 

Here's a video of the current best way to type a cheat on Android (keep in mind I'm sorta experienced with it at this point), sped up 2X for convenience: https://drive.google.com/file/d/18PTuQdElS7c2XsFSqbKmqVG-3gAp23Nk/view?usp=drivesdk

I will include a screenshot of the keyboard input I believe would be best fit, when I find an app that allows me to use it. Maybe something like an Android dialogue window would be good too.

**Comment by flyinghead** [comment]
The error message indicates "Can't save cheat file", probably because Flycast doesn't have permissions to write to its `data` folder. This explains why the cheat disappears on subsequent sessions since it's not saved to permanent storage. 

Regarding the UI, the keyboard is the native Android on-screen keyboard. The UI however isn't native. So there's some clunkiness when mixing the two. But Flycast running on so many platforms, I don't have the luxury to have a native UI for each one.

**Comment by Gonckzilla** [comment]
Yep! Moved the Flycast System Data from Downloads to Documents (default location), and my cheats appeared (I migrated phones not long ago). So yeah it appears to have been a storage access issue that somehow didn't affect the saving of settings. 

About the cheats UI, it's certainly understandable and reasonable that you wouldn't develop a device specific interface for each OS. What I had in mind is a much simpler implementation, but I'm having trouble explaining it and if I find an app that uses it, I'll take a video of it and show it. It looks something like this. 
![16737348956725007641088493231137](https://user-images.githubusercontent.com/70910647/212499681-2dab01a7-ee1d-4af8-8c3e-cf951521a8a5.png)
I've always had the feeling that this is an ease-of-use window that comes from the keyboard Android software itself, and simply as a text displayer which then sends to the app. I say this because in games that have their own UIs, I recall this window coming up and taking up the whole screen. 

However like I said, it's perfectly reasonable not to invest efforts in this OS-specific improvement. I only suggested it because I imagined it could be simpler given the existence of this keyboard window. I hope I communicate my point? (both in explaining the window thing, and in this last paragraph :s)
